Makefile

```diff
@@ -5,90 +5,77 @@ SHELL := bash
-
-export version
-
-upgrade: fetch down # Upgrade to the latest release without losing persistent data. Usage: `make upgrade version=v1.5.0` (version is optional)
-ifeq ($(version),)
-	version=$(shell curl --silent  "https://api.github.com/repos/minvws/nl-kat-coordination/tags" | jq -r '.[].name' | grep -v "rc" | head -n 1)
-	make upgrade version=$$version
-else
-	make checkout branch=$(version)
-	make build-all
-	make up
-endif
-
-reset:
-	-docker-compose down --remove-orphans --volumes --timeout 0
-	make up
-	make -C boefjes build
-	make -C rocky almost-flush
+# Bring down containers, remove all volumes, and bring them up again
+reset: clean kat
 
+# Bring up containers
 up:
-	docker-compose up -d --force-recreate
+	docker-compose up --detach
 
+# Bring down containers without data loss
 down:
 	-docker-compose down
 
+# Bring down containers and remove all volumes
+clean:
+	-docker-compose down --timeout 0 --volumes --remove-orphans
+
+# Fetch the latest changes from the Git remote
 fetch:
-	-git fetch
+	git fetch --all --prune --tags
 
+# Pull the latest changes from the default upstream
 pull:
-	-git pull
+	git pull
+
+# Upgrade to the latest release without losing persistent data. Usage: `make upgrade version=v1.5.0` (version is optional)
+VERSION?=$(shell curl -sSf "https://api.github.com/repos/minvws/nl-kat-coordination/tags" | jq -r '[.[].name | select(. | contains("rc") | not)][0]')
+upgrade: down fetch
+	git checkout $(VERSION)
+	make kat
 
+# Create .env file only if it does not exist
 env-if-empty:
 ifeq ("$(wildcard .env)","")
 	make env
 endif
 
-env:  # Create .env file from the env-dist with randomly generated credentials from vars annotated by "{%EXAMPLE_VAR}"
-	$(HIDE) cp .env-dist .env
+# Create .env file from the env-dist with randomly generated credentials from vars annotated by "{%EXAMPLE_VAR}"
+env:
+	cp .env-dist .env
+	echo "Initializing .env with random credentials"
 ifeq ($(UNAME), Darwin)  # Different sed on MacOS
 	$(HIDE) grep -o "{%\([_A-Z]*\)}" .env-dist | sort -u | while read v; do sed -i '' "s/$$v/$$(openssl rand -hex 25)/g" .env; done
 else
 	$(HIDE) grep -o "{%\([_A-Z]*\)}" .env-dist | sort -u | while read v; do sed -i "s/$$v/$$(openssl rand -hex 25)/g" .env; done
 endif
 
-checkout: # Usage: `make checkout branch=develop`
-	-git checkout $(branch)
-
-pull-reset:
-	-git reset --hard HEAD
-	-git pull
-
-build-all:  # Build should prepare all other services: migrate them, seed them, etc.
-ifeq ($(UNAME), Darwin)
-	docker-compose build --build-arg USER_UID="$$(id -u)"
+# Build will prepare all services: migrate them, seed them, etc.
+build:
+ifeq ($(UNAME),Darwin)
+	docker-compose build --pull --build-arg USER_UID="$$(id -u)"
 else
-	docker-compose build --build-arg USER_UID="$$(id -u)" --build-arg USER_GID="$$(id -g)"
+	docker-compose build --pull --build-arg USER_UID="$$(id -u)" --build-arg USER_GID="$$(id -g)"
```

Comment:
Now that we're looking into the performance, what do you think of building the images in parallel with `--parallel`, perhaps optionally? The output does become rather ugly.

Reply:
On my machine, Docker Compose (version v2.17.3) already builds the images in parallel. Can't really find when this became the default, but we could add `--parallel` explicitly for older Docker Compose versions.

Reply:
Interesting, I'll upgrade especially given this message from the docker documentation: 

![image](https://user-images.githubusercontent.com/46660228/235882914-7f8621da-4920-48e0-87d1-f28c1542c163.png)

Reply:
I've added `--parallel` explicitly, doesn't hurt.